source/service.go

```diff
@@ -148,6 +155,37 @@ func (sc *serviceSource) endpoints(svc *v1.Service) []*endpoint.Endpoint {
 	return endpoints
 }
 
+// filterByAnnotations filters a list of services by a given annotation selector.
+func (sc *serviceSource) filterByAnnotations(services []v1.Service, annotationFilter string) ([]v1.Service, error) {
```

Comment:
I think there was no case to pass the filter as parameter, because it's already member of serviceSource.